smart-contract/src/contract.rs

```diff
@@ -528,6 +536,105 @@ fn try_set_admin(
         .add_attribute("admin", &state.admin))
 }
 
+fn try_add_executor(
+    deps: DepsMut<ProvenanceQuery>,
+    info: MessageInfo,
+    id: String,
+) -> Result<Response<ProvenanceMsg>, ContractError> {
+    // Validate params.
+    if !info.funds.is_empty() {
+        return Err(contract_err("no funds should be sent during add executor"));
+    }
+
+    let address = deps.api.addr_validate(&id)?.into_string();
+    let mut state = config_read(deps.storage).load()?;
+
+    // Ensure message sender is admin.
+    if info.sender != state.admin {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Ensure executor wasn't already added
+    if state.executors.contains(&address) {
+        return Err(contract_err("executor already exists"));
+    }
+
+    // Add the executor and save
+    state.executors.push(address.clone());
```

Comment:
Nit: it looks like Response can take a `&str` so this should work if you wanted to avoid the need for clone.

```rust
let resp = Response::new()
    .add_attribute("action", "add_executor")
    .add_attribute("executor", &address));

state.executors.push(address);
config(deps.storage).save(&state)?;

Ok(resp)
```